src/core_plugins/elasticsearch/lib/__tests__/health_check.js

```diff
@@ -207,12 +207,17 @@ describe('plugins/elasticsearch', () => {
     });
 
     describe('#waitUntilReady', function () {
-      it('polls health until index is ready', function () {
+      it('polls health until index is ready, then waits for green status', function () {
         const clusterHealth = cluster.callWithInternalUser.withArgs('cluster.health', sinon.match.any);
         clusterHealth.onCall(0).returns(Promise.resolve({ timed_out: true }));
         clusterHealth.onCall(1).returns(Promise.resolve({ status: 'red' }));
         clusterHealth.onCall(2).returns(Promise.resolve({ status: 'green' }));
 
+        plugin.status.once = function (event, handler) {
+          expect(event).to.be('green');
```

Comment:
The problem with burying assertions like this inside a callback is that they don't fail if the callback isn't invoked at all.  If I do a refactor on the status lifecycle or something that results in once not being called, or in this case simply changing the name of the function, this whole test would continue to pass even though the status may not actually be green.